src/base-xhr.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+var objectMirrors = require('./object-mirrors');
+
+var BaseXHR = function(xhrImpl, onreadystatechange, onprogress, onloadend) {
```

Comment:
I don't understand why you need to make a special case out of it. Couldn't you treat them using `mirrorRwProp` as well?